modules/phase_field/src/materials/GrandPotentialSinteringMaterial.C

```diff
@@ -113,11 +119,16 @@ GrandPotentialSinteringMaterial::GrandPotentialSinteringMaterial(const InputPara
     _mu_gb(6.0 * _sigma_gb / _int_width),
     _kappa_s(0.75 * _sigma_s * _int_width),
     _kappa_gb(0.75 * _sigma_gb * _int_width),
-    _kB(8.617343e-5) // eV/K
+    _kB(8.617343e-5), // eV/K
+    _mass_conservation(getParam<bool>("mass_conservation"))
 {
   if ((_switch > 1.0) || (_switch < 0.0))
     mooseError("GrandPotentialSinteringMaterial: surface_switch_value should be between 0 and 1");
 
+  if ((_mass_conservation) && (_solid_energy > 0))
```

Comment:
FTI this would be fine, too (and preferred_

```suggestion
  if (_mass_conservation && _solid_energy > 0)
```